imcsdk/apis/server/serveractions.py

```diff
@@ -46,21 +49,42 @@ def power_up_server(handle):
         power_up_server(handle)
     """
 
+    import time
     from imcsdk.mometa.compute.ComputeRackUnit import ComputeRackUnit,\
         ComputeRackUnitConsts
 
-    rack_mo = ComputeRackUnit(parent_mo_or_dn="sys", server_id="1")
-    rack_mo.admin_power = ComputeRackUnitConsts.ADMIN_POWER_UP
-    handle.set_mo(rack_mo)
-    return rack_mo
-
-
-def power_down_server(handle):
+    # Turn power on only if not already powered up
+    if get_server_power_state(handle) != "on":
+        rack_mo = ComputeRackUnit(parent_mo_or_dn="sys", server_id="1")
+        rack_mo.admin_power = ComputeRackUnitConsts.ADMIN_POWER_UP
+        handle.set_mo(rack_mo)
+
+    # Poll until the server is powered up
+    wait_time = 0
```

Comment:
@britthouser This entire wait logic is sort of similar for power_up and power_down except for the state we have to check on.. can we abstract this in a internal method in the same file, `_wait_for_power_state(handle, "on"/"off")` and move all this logic there. some De-dup.